plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -61,6 +65,49 @@
 
     public static final String TYPE = "s3";
 
+    // prefix for aws client settings
+    private static final String PREFIX = "aws.config.";
```

Comment:
What about something like `s3.client.`? I think anything regarding the client configuration should stay under this named config, and anything else should be repository specific (eg the bucket). I actually don't think we need to have fallback settings like the other `bucket` or `server_side_encryption`. The client settings (ie the ones I have moved in this PR) should be under named configs, and `bucket`, etc should be in the repository settings.  Users just aren't setting these up that often, I don't see a real need for multiple levels of configuration (and the complication in code and testing that comes with it).